weasyl/define.py

```diff
@@ -671,6 +671,20 @@ def private_messages_unread_count(userid: int) -> int:
         "SELECT COUNT(*) FROM message WHERE otherid = %(user)s AND settings ~ 'u'", user=userid)
 
 
+def site_updates_unread_count(userid: int) -> int:
+    last_read_updateid = engine.scalar("""
+        SELECT COALESCE(updateid, 0)
+        FROM siteupdateread
+        WHERE userid = %(user)s
+    """, user=userid)
```

Comment:
Reuse `get_last_read_updateid`? But, since `get_last_read_updateid` exists, I’m going to suggest this alternative after all:

- `get_last_read_updateid` can be cached
- the list of all update ids in descending order can be cached (globally) and invalidated when a new site update is posted
- the count of unread site updates becomes `[*get_updateids(), None].index(get_last_read_updateid(userid))`, which can be added outside of `_page_header_info` for instant notifications and a reduced number of queries for each uncached `_page_header_info`